python/afdko/buildcff2vf.py

```diff
@@ -1,998 +1,486 @@
-        return
-
-    if len(args) == 2:
-        designSpacePath, varFontPath = args
-    elif len(args) == 1:
-        designSpacePath = args[0]
-        varFontPath = os.path.splitext(designSpacePath)[0] + '.otf'
+def add_glyph_names(tt_font, glyph_order):
+    postTable = tt_font['post']
+    postTable.glyphOrder = tt_font.glyphOrder = glyph_order
+    postTable.formatType = 2.0
+    postTable.extraNames = []
+    postTable.mapping = {}
+    postTable.compile(tt_font)
+
+
+def _validate_path(path_str):
+    # used for paths passed to get_options.
+    valid_path = os.path.abspath(os.path.realpath(path_str))
+    if not os.path.exists(valid_path):
+        raise argparse.ArgumentTypeError(
+            "'{}' is not a valid path.".format(path_str))
+    return valid_path
+
+
+def get_options(args):
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawTextHelpFormatter,
+        description=__doc__
+    )
+    parser.add_argument(
+        '--version',
+        action='version',
+        version=__version__
+    )
+    parser.add_argument(
+        '-v',
+        '--verbose',
+        action='count',
+        default=0,
+        help='verbose mode\n'
+             'Use -vv for debug mode'
+    )
+    parser.add_argument(
+        '-d',
+        '--designspace',
+        metavar='PATH',
+        dest='design_space_path',
+        type=_validate_path,
+        help='path to design space file',
+        required=True
+    )
+    parser.add_argument(
+        '-o',
+        '--output',
+        dest='var_font_path',
+        metavar='PATH',
+        help='path to output variable font file. Default is base name\n'
+        'of the design space file.',
+    )
+    parser.add_argument(
+        '-k',
+        '--keep-glyph-names',
+        action='store_true',
+        help='Preserve glyph names in output var font, with a post table\n'
+        'format 2.',
+    )
+    parser.add_argument(
+        '-c',
+        '--check-compat',
+        dest='check_compatibility',
+        action='store_true',
+        help='Check outline compatibility in source fonts, and fix flat\n'
+        'curves.',
+    )
+    parser.add_argument(
+        '-i',
+        '--include-glyphs',
+        dest='include_glyphs_path',
+        metavar='PATH',
+        type=_validate_path,
+        help='Path to file containing a python dict specifying which\n'
+        'glyph names should be included from which source fonts.\n'
+    )
+    options = parser.parse_args(args)
+
+    if not options.var_font_path:
+        var_font_path = '{}.otf'.format(os.path.splitext(
+            options.design_space_path)[0])
+        options.var_font_path = var_font_path
+
+    if not options.verbose:
+        level = PROGRESS_LEVEL
+        logging.basicConfig(level=level, format="%(message)s")
     else:
-        print(__usage__)
-        return
+        if options.verbose:
+            level = logging.INFO
+        logging.basicConfig(level=level)
```

Comment:
The `level` might not be set as of this line or the next. Perhaps set `level` to some default before line 430 (i.e. the start of the `if...else...` block, or else sort this out via indentation.

Reply:
Must have left this behind when editing logic. The issue is actually that the test 'if options.verbose' is redundant, as it alway true here.

Reply:
OK, I see that this is also fixed now, so resolving this conversation too.